### ISSUE TITLE
`#to_hash` being called on `initialize_with` `new` arguments

### ISSUE THREAD

**Issue by dflynn15** [thoughtbot/factory_bot#1473]
### Description

This is going to sound... odd. Also, before reading ahead it is important to note that this appears to _only_ be a Ruby 2.7 issue!

**Given:**
- I have created a factory using `initialize_with` using `new` and not `DescribedClass.new`
- `DescribedClass` has more than one argument and there exists an optional argument

**Then:**
- `#to_hash` is called on arguments _past_ the first argument.

### Reproduction Steps
```ruby
require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"
  gem "factory_bot", "~> 6.0"
end

require "factory_bot"
require "minitest/autorun"
require "logger"

class Post
  attr_reader :name, :body, :tags

  def initialize(name, body, tags = nil)
    @name = name
    @body = body
    @tags = tags
  end
end

class BodyText
  def to_hash
    raise 'I should not be called'
  end
end

FactoryBot.define do
  factory :post do
    name { 'Person name' }
    body { BodyText.new }
    initialize_with { new(name, body) }
  end
end

class FactoryBotTest < Minitest::Test
  def test_factory_bot_stuff
    # expected
    FactoryBot.build(:post)

    assert_equal post.name, 'Person name'
  end
end
```

### Expected behavior

I do not expect any arguments passed into `new` to have `#to_hash` called on them.

### Actual behavior

`#to_hash` is being called on arguments being passed in initializers.

### System configuration
**factory_bot version**:  6.1.0
**rails version**:  6.0.3.5
**ruby version**: 2.7.2

**Comment by JoeDupuis** [comment]
Some clarifications. It appears to happen only on the last non optional argument if there is an omitted optional argument after it. 
In this case omitting `tags` will cause to_hash to be call on `body`.

**Comment by composerinteralia** [comment]
Ah yes, I think this is fixed by 793a651, but hasn't been released yet. It has to do with changes to keyword arguments in Ruby 2.7 and 3.0. We'll send out a release soon, but in the meantime you could point your Gemfile to that commit to get this working.
